src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.cpp

```diff
@@ -308,3 +419,17 @@ void BallPlacementPlayFSM::setupMoveTactics(const Update &event)
             TbotsProto::ObstacleAvoidanceMode::SAFE);
     }
 }
+
+bool BallPlacementPlayFSM::ballReleased(const BallPlacementPlayFSM::Update &event) {
+    Point ball_pos = event.common.world_ptr->ball().position();
+    std::optional<Robot> robot_placing_ball = event.common.world_ptr->friendlyTeam().getNearestRobot(ball_pos);
+
+    if (!robot_placing_ball.has_value())
+    {
+        return false;
+    }
+
+    return distance(ball_pos, robot_placing_ball->position()) > 0.4;
```

Comment:
you can reuse `ball_is_kicked` threshold in dynamic params: https://github.com/UBC-Thunderbots/Software/blob/907a935537592f8d6d238b684e958f7d67fcc1e8/src/proto/parameters.proto#L58

Reply:
Isnt ball_is_kicked a speed? This is a distance check

Reply:
Oh yeah you're right. Make `0.4` a constant. `BALL_PLACED_THRESHOLD_M`